### PR TITLE
inputplug: init at 0.4.0

### PR DESCRIPTION
**Pull request by jecaro**
###### Motivation for this change

[inputplug](https://github.com/andrewshadura/inputplug) is a very simple daemon which monitors XInput events and runs arbitrary scripts on hierarchy change events (such as a device being attached, removed, enabled or disabled).

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/inputplug/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ libbsd ];
+
+  cargoSha256 = "1j38354n318ap3akbjma0milgngyj3rlhkziq1mh8p8c8p0x0van";
+
+  meta = with lib; {
+    description = "Monitor XInput events and run arbitrary scripts on hierarchy change events";
+    homepage = "https://github.com/andrewshadura/inputplug";
+    license = [ licenses.mit ];
```
Comment (figsoda) [inline pkgs/tools/X11/inputplug/default.nix:29]:
```suggestion
    license = licenses.mit;
```

File: pkgs/tools/X11/inputplug/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    maintainers = [ maintainers.jecaro ];
```
Comment (figsoda) [inline pkgs/tools/X11/inputplug/default.nix:30]:
```suggestion
    maintainers = with maintainers; [ jecaro ];
```

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -4927,6 +4927,12 @@
     github = "jduan";
     githubId = 452450;
   };
+  jecaro = {
```
Comment (figsoda) [inline maintainers/maintainer-list.nix:4930]: this should be a separate commit: `maintainers: add jecaro`

File: pkgs/tools/X11/inputplug/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "andrewshadura";
+    repo = pname;
+    rev = version;
+    sha256 = "02pslkjrc20wjc8zr7fck9djbhyahh25xlva7pcabyp4jlh2kf8f";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
```
Comment (figsoda) [inline pkgs/tools/X11/inputplug/default.nix:19]: needs a patch comment

**Review by figsoda — CHANGES_REQUESTED** [review]
upstream includes a man page, use `installManPage` from `installShellFiles`

you can use `fetchCrate` to fetch from crates.io and remove the patch

File: pkgs/tools/X11/inputplug/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/tools/X11/inputplug/default.nix:19]: Please use the cargo.lock directly instead of using a patch.

Comment (figsoda) [reply pkgs/tools/X11/inputplug/default.nix:19]: It shouldn't be needed if you use `fetchCrate`, the crates.io version includes a [`Cargo.lock`](https://docs.rs/crate/inputplug/0.4.0/source/Cargo.lock)

**Comment by jecaro** [comment]
Thx for these comments. I updated the branch accordingly.

File: pkgs/tools/X11/inputplug/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "inputplug";
+  version = "0.4.0";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "00gv2i2pxy56l6ysslbscxinr4r0mpk9p2ivkrnjnwhc8j3v8v7h";
+  };
+
+  nativeBuildInputs = [ installShellFiles pkg-config ];
```
Comment (figsoda) [inline pkgs/tools/X11/inputplug/default.nix:18]:
```suggestion
  nativeBuildInputs = [ installShellFiles pkg-config ];

```

File: pkgs/tools/X11/inputplug/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  buildInputs = [ libbsd ];
+
+  cargoHash = "sha256:161kz47d4psfvh0vm98k8qappg50lpsw1ybyy7s3g3bp6ivfz8jv";
```
Comment (figsoda) [inline pkgs/tools/X11/inputplug/default.nix:21]: change to `cargoSha256`

File: pkgs/tools/X11/inputplug/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+  postInstall = "installManPage inputplug.1";
+
+  meta = with lib; {
+    description = "Monitor XInput events and run arbitrary scripts on hierarchy change events";
+    homepage = "https://github.com/andrewshadura/inputplug";
+    license = with licenses; [ mit ];
```
Comment (figsoda) [inline pkgs/tools/X11/inputplug/default.nix:28]:
```suggestion
    license = licenses.mit;
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/X11/inputplug/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/tools/X11/inputplug/default.nix:21]:
```suggestion
  cargoHash256 = "161kz47d4psfvh0vm98k8qappg50lpsw1ybyy7s3g3bp6ivfz8jv";
```

File: pkgs/tools/X11/inputplug/default.nix [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    inherit pname version;
+    sha256 = "00gv2i2pxy56l6ysslbscxinr4r0mpk9p2ivkrnjnwhc8j3v8v7h";
+  };
+
+  nativeBuildInputs = [ installShellFiles pkg-config ];
+  buildInputs = [ libbsd ];
+
+  cargoHash = "sha256:161kz47d4psfvh0vm98k8qappg50lpsw1ybyy7s3g3bp6ivfz8jv";
+
+  postInstall = "installManPage inputplug.1";
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/inputplug/default.nix:23]:
```suggestion
  postInstall = ''
    installManPage inputplug.1
  '';
```

File: pkgs/tools/X11/inputplug/default.nix (hunk 6 above)
Comment (figsoda) [reply pkgs/tools/X11/inputplug/default.nix:21]: it should be `cargoSha256` not `cargoHash256`

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 134949` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>inputplug</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Comment by jecaro** [comment]
Hopefully, it should be ok this time.
